hrsi_velocity_costmaps/scripts/velocity_costmap_server.py

```diff
@@ -23,24 +23,31 @@
 
 class VelocityCostmapServer(object):
     def __init__(self, name):
-        rospy.loginfo("Starting %s..." % name)
-        self.vis_pub = rospy.Publisher("~visualization_marker", Marker, queue_size=1)
+        rospy.loginfo("["+rospy.get_name()+"] " + "Starting ... ")
+
+        self.vis_marker_topic = rospy.get_param("~vis_marker_topic","~visualization_marker")
+        self.vel_costmap_topic = rospy.get_param("~vel_costmap_topic","/velocity_costmap")
+        self.origin_topic = rospy.get_param("~origin_topic","origin")
+        self.base_link_tf = rospy.get_param("~base_link_tf","base_link")
+
+        self.vis_pub = rospy.Publisher(self.vis_marker_topic, Marker, queue_size=1)
         self.tf = TransformListener()
         self.cc = CostmapCreator(
-            rospy.Publisher("/velocity_costmap", OccupancyGrid, queue_size=10, latch=True),
-            rospy.Publisher("~origin", PoseStamped, queue_size=10)
+            rospy.Publisher(self.vel_costmap_topic, OccupancyGrid, queue_size=10, latch=True),
+            rospy.Publisher(self.origin_topic, PoseStamped, queue_size=10)
         )
         self.dyn_srv = DynServer(VelocityCostmapsConfig, self.dyn_callback)
         subs = [
             Subscriber(rospy.get_param("~qtc_topic", "/qtc_state_predictor/prediction_array"), QTCPredictionArray),
             Subscriber(rospy.get_param("~ppl_topic", "/people_tracker/positions"), PeopleTracker)
         ]
-        ts = TimeSynchronizer(
+        self.ts = ApproximateTimeSynchronizer(
```

Comment:
The timestamps of the positions of people and the prediction array should be exactly the same as they are pulled through. Why the approximate time synchroniser?

Reply:
Not 100% sure about this, but in my simulation timestamps weren't the same, I had an small delay between them, so it wasn't working at all.  Maybe I miss configured something. Is it too bad to keep that change?

Reply:
If it works for you it should be fine but I am not sure why they would be different because everything uses the ppl tracker timestamp. Or should.